src/LockIn-LakeShore_M81-CM-10/main.py

```
@@ -417,12 +419,16 @@ def check_device(self):
-        if not self.lia_ref in ["S1", "S2", "S3"]:
-            if self.lia_ref.lower().startswith("reference"):
-                self.lia_ref = "RIN"  # External Reference Source
+        # The converted value is kept in a local variable. Overwriting self.lia_ref would
+        # make a second call of this function fail, because "RIN" is neither one of the
+        # source channels nor does it start with "reference".
+        reference = self.lia_ref
+        if not reference in ["S1", "S2", "S3"]:
+            if reference.lower().startswith("reference"):
+                reference = "RIN"  # External Reference Source
             else:
                 raise ValueError(f"No valid reference source selected: {self.lia_ref}.")
```

Ruff (0.16.3)

[error] Exception must not use an f-string literal, assign to variable first

Assign to variable; remove f-string literal

(EM102)